math_opt_benchmark/facility/ufl.cc

```diff
@@ -141,4 +156,122 @@ BenchmarkInstance UFLSolver::GetModel() {
   return instance_;
 }
 
+//
+// UFLBenders
+//
+
+UFLBenders::UFLBenders(const UFLProblem &problem)
+    : problem_(problem),
+    solver_(math_opt::SolverType::kGurobi, problem, true),
+    cost_indices_(problem.num_customers, std::vector<int>(problem.num_facilities)) {
+  for (int i = 0; i < problem_.num_customers; i++) {
+    std::vector<double>& costs = problem_.supply_costs[i];
+    std::vector<int>& indices = cost_indices_[i];
+    std::iota(indices.begin(), indices.end(), 0);
+    std::sort(indices.begin(), indices.end(), [&, costs](int i, int j) { return costs[i] < costs[j]; });
+    std::sort(costs.begin(), costs.end(), [](double i, double j) { return i < j; });
+  }
+}
+
+UFLSolution UFLBenders::benders() {
+  int num_customers = problem_.num_customers;
+  int num_facilities = problem_.num_facilities;
+  std::vector<std::vector<double>>& supply_costs = problem_.supply_costs;
+  UFLSolution solution = solver_.Solve();
+  double prev_objective = -1.0;
+  // This seems to work even though it's not ub >= lb
+  while (solution.objective_value != prev_objective) {
```

Comment:
this is a weird termination criteria. Can we instead monitor the difference between the best objective we have bound and the best bound (which is the bound at the current iteration)